internal/operators/lvm/lvm_operator.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/conversions"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/sirupsen/logrus"
+)
+
+// operator is an ODF LVM OLM operator plugin; it implements api.Operator
+type operator struct {
+	log       logrus.FieldLogger
+	config    *Config
+	extracter oc.Extracter
+}
+
+var Operator = models.MonitoredOperator{
+	Name:             "lvm",
+	OperatorType:     models.OperatorTypeOlm,
+	Namespace:        "openshift-storage",
+	SubscriptionName: "odf-lvm-operator",
+	TimeoutSeconds:   30 * 60,
+}
+
+// NewLvmOperator creates new LvmOperator
+func NewLvmOperator(log logrus.FieldLogger, extracter oc.Extracter) *operator {
+	cfg := Config{}
+	err := envconfig.Process(common.EnvConfigPrefix, &cfg)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	return newLvmOperatorWithConfig(log, &cfg, extracter)
+}
+
+// newOdfOperatorWithConfig creates new ODFOperator with given configuration
+func newLvmOperatorWithConfig(log logrus.FieldLogger, config *Config, extracter oc.Extracter) *operator {
+	return &operator{
+		log:       log,
+		config:    config,
+		extracter: extracter,
+	}
+}
+
+// GetName reports the name of an operator this Operator manages
+func (o *operator) GetName() string {
+	return Operator.Name
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []string {
+	return make([]string, 0)
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDLvmRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDLvmRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(_ context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	if !common.IsSingleNodeCluster(cluster) {
+		message := "ODF LVM operator is only supported for Single Node Openshift deployment"
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{message}}, nil
+	}
+
+	var ocpVersion, minOpenshiftVersionForLvm *version.Version
+	var err error
+
+	ocpVersion, err = version.NewVersion(cluster.OpenshiftVersion)
+	if err != nil {
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
+	}
+	minOpenshiftVersionForLvm, err = version.NewVersion(o.config.LvmMinOpenshiftVersion)
+	if err != nil {
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
+	}
+	if ocpVersion.LessThan(minOpenshiftVersionForLvm) {
+		message := "ODF LVM operator is only supported for openshift versions 4.11.0 and above"
```

Comment:
Use o.config.LvmMinOpenshiftVersion rather than hard-coding 4.11.0 in the string